server/src/main/java/org/elasticsearch/transport/TransportKeepAlive.java

```diff
@@ -160,7 +153,11 @@ void removeChannel(TcpChannel channel) {
         }
 
         @Override
-        protected void doRunInLifecycle() {
+        protected void doRun() throws Exception {
+            if (isClosed) {
+                return;
```

Comment:
nit One difference from the previous code I see is a 

```
logger.trace("lifecycle is stopping. exiting");
```

at this corresponding point. Not sure whether it's at all useful to have a similar log here, but just mentioning it in case you think it's useful to add one.

Reply:
++ I don't think this is useful. Note that there's no such log in the `onAfterInLifecycle` so we don't even always get a log when the component shuts down today.